engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/hydration/batch/NadelBatchHydrationObjectIdFieldBuilder.kt

```diff
@@ -0,0 +1,132 @@
+package graphql.nadel.enginekt.transform.hydration.batch
+
+import graphql.nadel.Service
+import graphql.nadel.enginekt.blueprint.NadelBatchHydrationFieldInstruction
+import graphql.nadel.enginekt.blueprint.NadelOverallExecutionBlueprint
+import graphql.nadel.enginekt.blueprint.hydration.NadelBatchHydrationMatchStrategy
+import graphql.nadel.enginekt.transform.artificial.NadelAliasHelper
+import graphql.nadel.enginekt.util.resolveObjectTypes
+import graphql.nadel.enginekt.util.unwrap
+import graphql.normalized.ExecutableNormalizedField
+import graphql.normalized.ExecutableNormalizedField.newNormalizedField
+import graphql.schema.GraphQLFieldDefinition
+import graphql.schema.GraphQLNamedType
+import graphql.schema.GraphQLOutputType
+import graphql.schema.GraphQLType
+
+/**
+ * Builds the field used to identify objects returned by batch hydration e.g.
+ *
+ * ```graphql
+ * type Query {
+ *   issues(ids: [ID]): [Issue]
+ * }
+ *
+ * type Issue {
+ *   id: ID
+ *   relatedIssueIds: [ID]
+ *   key: String
+ *
+ *   relatedIssues: [Issue] @hydrated(
+ *      service: "IssueService"
+ *      field: "issues"
+ *      arguments : [{ name: "ids" value: "$source.relatedIssueIds" }]
+ *      identifiedBy: "id"
+ *      batchSize: 2
+ *   )
+ * }
+ * ```
+ *
+ * And given a hydration query
+ *
+ * ```graphql
+ * query {
+ *   issues(ids: ["PROJ-1", "PROJ-5", "PROJ-3"]) {
+ *      key
+ *   }
+ * }
+ * ```
+ *
+ * We need an id to actually identify the objects to know where to insert them back
+ * into the original result.
+ *
+ * i.e. this code inserts the selection for the `id` field
+ */
+internal object NadelBatchHydrationObjectIdFieldBuilder {
```

Comment:
Moved this code to another class because it was bloating original class with random concerns WRT field buildin

Reply:
I had to edit this function because originally it sent the underlying type name, but we need to be setting the overall type name